### PR TITLE
Admin uploads view

### PR DESCRIPTION
**Pull request by tannermpeterson**

### CODE REVIEW

File: frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js [hunk 1]
```diff
@@ -173,88 +173,48 @@ export default function AnalysisParamForm({
...
   const updateParams = (newParams) => {
-    setAllErrorMessagesBlank();
     const updatedParams = { ...analysisParams, ...newParams };
 
-    if (newParams.twitchWidths) {
+    if ("twitchWidths" in newParams) {
       validateTwitchWidths(updatedParams);
     }
-    if (newParams.startTime || newParams.endTime) {
+    if ("startTime" in newParams || "endTime" in newParams) {
```
Comment (luciipak) [inline frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js:181]: this is cool, I forgot you could do this with javascript

File: frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js [hunk 2]
```diff
@@ -173,88 +173,48 @@ export default function AnalysisParamForm({
...
+    for (const paramName of [
+      "prominenceFactorPeaks",
+      "prominenceFactorValleys",
+      "widthFactorPeaks",
+      "widthFactorValleys",
+      "maxY",
+    ]) {
+      if (paramName in newParams) {
+        validatePositiveNumber(updatedParams, paramName, false);
+      }
```
Comment (luciipak) [inline frontend/pulse3d/src/components/uploadForm/AnalysisParamForm.js:194]: 👍🏽

File: frontend/pulse3d/src/serviceWorker.js [hunk 3]
```diff
@@ -33,6 +31,7 @@ const tokens = {
 const setTokens = ({ access, refresh }) => {
   tokens.access = access.token;
   tokens.refresh = refresh.token;
+  setLogoutTimer();
```
Comment (luciipak) [inline frontend/pulse3d/src/serviceWorker.js:34]: so this is where I was saying just clears the timeout and doesn't ensure that the `ClientSource.postMessage({ logout: true });` gets called inside the logout. Do you think that's something we should worry about or no?

Comment (tannermpeterson) [reply frontend/pulse3d/src/serviceWorker.js:34]: I think `setLogoutTimer` does that, do you mean when the timer gets cleared?

Comment (luciipak) [reply frontend/pulse3d/src/serviceWorker.js:34]: Yes. I'll try to investigate the order of requests made next time it happens, but clearing the timeout doesn't ensure that the client enforces the auto logout

Comment (luciipak) [reply frontend/pulse3d/src/serviceWorker.js:34]: and the only time the logout message gets sent to _app.js is inside the timeout callback

Comment (luciipak) [reply frontend/pulse3d/src/serviceWorker.js:34]: Also, my bad, I meant to addd this comment to:
`const clearTokens = () => {
  clearLogoutTimer();
};`

**Review by luciipak — APPROVED** [review]

**Pull request merged**